pkg/segment/writer/agiletree.go

```diff
@@ -216,7 +216,7 @@ func (stb *StarTreeBuilder) DropSegTree(stbDictEncWorkBuf [][]string) {
 	stb.ResetSegTree(stb.groupByKeys, stb.mColNames, stbDictEncWorkBuf)
 }
 
-func (stb *StarTreeBuilder) DropColumns(colsToDrop []string) error {
+func (stb *StarTreeBuilder) DropColumns(colsToDrop map[string]uint64) error {
```

Comment:
What is the `uint64` for in this map? It doesn't look like you use the `uint64` in this function, so you probably shouldn't have it.

It looks like this is actually the map of column names and their cardinality that you're getting like this:
```
colsToDrop := segstore.getColsAboveCardLimit()
```
Is that right?

Here's what I would suggest:
1. Find a one-liner way to get the keys from a map. Getting them either as a `[]string` or a `map[string]struct{}` should be fine. Write a util in `maputils.go` if you need to
2. Change `colsToDrop` either back to `[]string` or to `map[string]struct{}`
3. Keep `colsToDrop := segstore.getColsAboveCardLimit()`, but then call your one-liner to extract the keys as a slice/set, and pass that to `DropColumns()`

Reply:
addressed in the latest commit.